src/core.c/Sequence.pm6

```diff
@@ -39,8 +44,8 @@ Routine.'!configure_positional_bind_failover'(Positional, PositionalBindFailover
 
 my role Sequence does PositionalBindFailover {
     multi method Array(::?CLASS:D:) { Array.from-iterator(self.iterator) }
-    multi method List(::?CLASS:D:)  { List.from-iterator(self.iterator) }
-    multi method Slip(::?CLASS:D:)  { Slip.from-iterator(self.iterator) }
+    multi method List(::?CLASS:D:)  { self.list.List }
```

Comment:
Feels to me the `.List` is a noop at that stage?

Reply:
I would actually propose to make the `List` method leading, and `list` be a frontend for `List`.  Then, copy the contents of the `List` method, and change `$!list` to `p6bindattrvinres(nqp::create(Slip),List,'$!list`)` and `List.from-iterator` to `Slip.from-iterator`.

Reply:
@lizmat, `.list.List` is indeed equivalent to `.list.self` _if_ `.list` returns a `List`. However, a `PositionalBindFailover` implementation may define `.list` to return a different type of list in which case `.List` is required to actually coerce the return value to `List`.

Regarding your suggestions, I consider them to be outside the scope of this PR as I just want to avoid unnecessary copying of `List`s. But I do have an opinion about them.

As I understand it, `PositionalBindFailover` is intended to be a minimalistic role that can be used by classes outside the core. The `.list` method is part of it because it's the method called by many built-ins that require a list-like representation of an object. Examples of such built-ins are `Any`'s `.List`, `.Slip` and `.Array` coercers. In fact, the `.Slip` implementation provided by `Sequence` is now exactly the same as the one provided by `Any`. I don't think that means the implementation can be removed from `Sequence`, because role `Sequence` does not carry the `Any` class.

Regarding the copying of implementations, I think such a violation of DRY is permissible only if it has noticeable performance benefits in real-world use cases. I've decided to keep identifying use cases and measuring performance outside the scope of this PR.